aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -0,0 +1,1081 @@
+        let owners_removed = vector::empty<address>();
+        while (i < len) {
+            let owner_to_remove = *vector::borrow(&owners_to_remove, i);
+            let (found, index) = vector::index_of(owners, &owner_to_remove);
+            // Only remove an owner if they're present in the owners list.
+            if (found) {
+                vector::push_back(&mut owners_removed, owner_to_remove);
+                vector::swap_remove(owners, index);
+            };
+            i = i + 1;
+        };
+
+        // Make sure there's still at least as many owners as the number of signatures required.
+        // This also ensures that there's at least one owner left as signature threshold must be > 0.
+        assert!(
+            vector::length(owners) >= multisig_account_resource.signatures_required,
+            error::invalid_state(ENOT_ENOUGH_OWNERS),
+        );
+
+        emit_event(&mut multisig_account_resource.remove_owners_events, RemoveOwnersEvent { owners_removed });
+    }
+
+    /// Update the number of signatures required to execute transaction in the specified multisig account.
+    /// This can only be invoked by the multisig account itself, through the proposal flow.
+    public entry fun update_signatures_required(
+        multisig_account: &signer, new_signatures_required: u64) acquires MultisigAccount {
+        let multisig_address = address_of(multisig_account);
+        assert_multisig_account_exists(multisig_address);
+        let multisig_account_resource = borrow_global_mut<MultisigAccount>(multisig_address);
+        // Short-circuit if the new number of signatures required is the same as before.
+        // This avoids emitting an event.
+        if (multisig_account_resource.signatures_required == new_signatures_required) {
+            return
+        };
+        let num_owners = vector::length(&multisig_account_resource.owners);
+        assert!(
+            new_signatures_required > 0 && new_signatures_required <= num_owners,
+            error::invalid_argument(EINVALID_SIGNATURES_REQUIRED),
+        );
+
+        let old_signatures_required = multisig_account_resource.signatures_required;
+        multisig_account_resource.signatures_required = new_signatures_required;
+        emit_event(
+            &mut multisig_account_resource.update_signature_required_events,
+            UpdateSignaturesRequiredEvent {
+                old_signatures_required,
+                new_signatures_required,
+            }
+        );
+    }
+
+    /// Create a multisig transaction, which will have one approval initially (from the creator).
+    ///
+    /// @param target_function The target function to call such as 0x123::module_to_call::function_to_call.
+    /// @param args Vector of BCS-encoded argument values to invoke the target function with.
+    public entry fun create_transaction(
+        owner: &signer,
+        multisig_account: address,
+        target_function: String,
+        args: vector<u8>,
```

Comment:
Should a txid be passed in by the user? This ensures that the transactions are in a determined order.

Reply:
The transaction id is an incrementing id. Having users pass in a custom transaction id doesn't quite make sense as it's managed/tracked by the module and can lead to edge cases + extra required validations.

Reply:
What I'm thinking is that, like the nonce in Ethereum transactions, it's incremental, but still need to specify the nonce to make sure that the transactions are created in a certain order.

Reply:
The current implementation enforce in-order execution of all multisig transactions created via this module. Is there a use case you're thinking of that needs this nonce to be specified and is not supported by in-order execution here?